networking_aci/plugins/ml2/drivers/mech_aci/driver.py

```diff
@@ -203,12 +204,28 @@ def _get_clearable_phys_doms(self, network, local_segment, host_config):
             other_bindings = (session.query(models.PortBindingLevel.host, models.PortBindingLevel.segment_id)
                                      .join(segment_model.NetworkSegment,
                                            segment_model.NetworkSegment.id == models.PortBindingLevel.segment_id)
+                                     .join(models.PortBinding,
+                                           models.PortBindingLevel.port_id == models.PortBinding.port_id)
                                      .filter(segment_model.NetworkSegment.network_id == network['id'],
                                              models.PortBindingLevel.level == 1,
                                              models.PortBindingLevel.segment_id != local_segment['id'])
                                      .distinct())
 
         for other_binding in other_bindings:
+            host = other_binding.host
+
+            # try to get alternate host from binding profile
+            binding_profile = None
+            try:
+                binding_profile = json.loads(other_binding.profile)
+                if binding_profile:
+                    switch = CiscoACIMechanismDriver.switch_from_local_link(binding_profile)
+                    if switch:
+                        host = host
+            except json.decoder.JSONDecodeError:
+                # don't act on broken binding profile
+                pass
+
             _, other_binding_host_config = self._host_or_host_group(other_binding.host)
```

Comment:
Do you need to use `host` here, too?

Reply:
Yes. The `host = host` apparently prevented flake8 from warning me.